File: lib_arch/src/main/java/com/yyxnb/arch/annotations/BarStyle.java
package com.yyxnb.arch.annotations;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({BarStyle.DarkContent, BarStyle.LightContent, BarStyle.None})
@Retention(RetentionPolicy.SOURCE)
public @interface BarStyle {

    // 深色
    int DarkContent = 1;
    // 浅色
    int LightContent = 2;
    // none
    int None = 0;
}

File: lib_arch/src/main/java/com/yyxnb/arch/annotations/BarStyle.kt
package com.yyxnb.arch.annotations

import androidx.annotation.IntDef

/**
 * 状态栏文字颜色
 *
 * @author yyx
 */
@IntDef(BarStyle.DARK_CONTENT, BarStyle.LIGHT_CONTENT, BarStyle.NONE)
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class BarStyle {
    companion object {
        /**
         * 深色
         */
        const val DARK_CONTENT = 1

        /**
         * 浅色
         */
        const val LIGHT_CONTENT = 2

        /**
         * none
         */
        const val NONE = 0
    }
}